tests/rptest/services/chaos/faults/isolate_leader.py

```diff
@@ -0,0 +1,42 @@
+# Copyright 2024 Redpanda Data, Inc.
+#
+# Use of this software is governed by the Business Source License
+# included in the file licenses/BSL.md
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0
+
+from rptest.services.admin import Admin
+from rptest.services.failure_injector import FailureInjector
+
+from .fault import RecoverableFault
+
+
+class IsolateLeaderFault(RecoverableFault):
+    def __init__(self,
+                 redpanda,
+                 topic: str,
+                 partition: int = 0,
+                 namespace: str = "kafka",
+                 name: str = "isolate_leader"):
+        self._name = name
+        self.redpanda = redpanda
+        self.topic = topic
+        self.partition = partition
+        self.namespace = namespace
+
+        self._leader = None
+
+    @property
+    def name(self):
+        return self._name
+
+    def inject(self):
```

Comment:
nit: curious if these can be implemented with context manager.

Reply:
Thought about it, but probably not worth it, as it is used in one place so explicit finally is not onerous.